### DISCUSSION TITLE
latest release is working great on my 2020 ID5 mini

### DISCUSSION THREAD

**Discussion by dfortis** [BimmerGestalt/AAIdrive#326] [General]
You saved my 2020's ID5 from slow death! great job! Spotify, notifications (even images on WhatsApp notifications!), everything is working fine. Strangely i'm running latest Spotify and integration is still working. A coffee from me :)

**Comment by hufman** [comment]
I'm so glad to hear that! I'm a Mini fan too :)

**Reply by sfortis** [reply]
@hufman reading your documents i understood that you can grab the keys from bmw/mini app. Cant you integrate them into androidautoidrive and get rid of the bmw/mini app? i mean, how hard is to establish direct connection with the car?

**Reply by hufman** [reply]
The biggest obstacle is just because it's easier to reuse the existing connection process and build functionality on top of it.
The actual connection handling code wouldn't be too hard to replace, it's a relatively simple multiplexing protocol on top of the Bluetooth or USB serial connection, with Android providing a clean Socket abstraction over each. There would be a tricky process to make sure the official car app isn't trying to manage the connection at the same time.
The hardest part is how to handle the security challenge from the car, which I believe involves a cryptographic signature from an RSA key. I haven't figured out how to extract this key, but I feel the least confident about the legality of extracting or redistributing this key.
Since most BMW drivers would also have the BMW app installed, I decided against trying to replace that app and to just build on top of it.

**Comment by donsaimon** [comment]
How to have whatsapp notifications?? I have a 3 series BMW with ID5

**Reply by dfortis** [reply]
On my mini i just enabled notifications from the app. Every phone notification is now pushed into to car. On IM notifications i can also answer using the on screen keyboard.

**Reply by donsaimon** [reply]
Wow! In AAI settings I found only 'sms" enable... But they do not "ring" on the car... What do you mean with "IM" notifications?

**Reply by dfortis** [reply]
great ! i mean notifications coming from instant messaging apps which provide the "reply" field in their context.

**Reply by hufman** [reply]
With the Notifications feature enabled, you should see a new icon in the car, both an empty book under Connected Drive and the same book with the label Notifications in the Office menu: https://github.com/hufman/AndroidAutoIdrive/blob/master/app/src/main/res/drawable/pic_news_ambutton_bmw_id5.jpg
Any WhatsApp notification showing on the phone should show up in this list, new notifications should also show up in the car's Message Center, popups and sound over Bluetooth are configurable.

**Reply by donsaimon** [reply]
How to configure popups and sound over Bluetooth? Via AAI app? Where?

**Reply by hufman** [reply]
You should see a full set of toggle switches in the Notification tab after enabling the feature, and a similar set of checkboxes in the car itself when viewing the notification list through that book icon.

**Comment by donsaimon** [comment]
Ok perfect, found! Thanks a lot!

**Comment by donsaimon** [comment]
How may I increase notification sound (I mean the "ring", not the text reading voice) volume?
Moreover, I have noticed that, when I receive a notification, sometimes automatically the page of notifications is opened, sometimes I only see the message icon on the top right of the display and I have to manually open the notifications page.

**Reply by hufman** [reply]
There isn't a way to increase the notification sound volume, it plays through the normal Media stream which is maxed out by default already.
Notification popups are suppressed for a few seconds after interacting with the IDrive system, to not interrupt the user, which will then show just the envelope. This is mentioned in the Tips panel in the Notification tab, which also suggests some other helpful tips like saving a shortcut button to the Notification app for easy access.

**Reply by donsaimon** [reply]
Ok Walter thanks for the explanation. I will check if the notifications page appears only if I am not interacting with iDrive controller.

**Comment by Krakuanen** [comment]
Works perfectly on my 2018 Mini Clubman as well. The car has very basic media functions (no navigation or Carplay. Not even Mini connected is supported when I enter the VIN). Amazing app! Sound over USB also works, even though the app says that it's not supported (forced sound over USB in advanced settings). Only thing that is missing now is Google maps....

**Reply by hufman** [reply]
I'm so glad to hear that! That's awesome that the audio works, what phone and Android version do you have? The logic for detecting support is based entirely on Android version, and I wonder how your phone is able to handle it better.

**Comment by Krakuanen** [comment]
Hi! I have an old Samsung Galaxy S7 running version 8.0. The car has ID5
and EntryEvo. I contacted Bimmer, they said no new functions could be added
with out an hardware upgrade. But they have probably not heard of your app!
The app can connect via BT as well, but USB is very stable. I'm sure audio
is routed through USB, as it works even when I disable BT on my phone.
Maybe something has changed on the latest Nightly build, because sound over
USB didn't work before (could also be some other reason). I'm really
impressed by your work, I hope more people will discover your app!

Den mån 26 juli 2021 20:01Walter Huf ***@***.***> skrev:
